packages/stems/src/components/Scrollbar/Scrollbar.tsx

```diff
@@ -15,64 +15,71 @@ import { ScrollbarProps } from './types'
  * `Scrollbar` uses react-perfect-scrollbar (https://www.npmjs.com/package/react-perfect-scrollbar)
  * under the hood. For advanced use cases, refer to the documentation.
  */
-export const Scrollbar = ({
-  children,
-  className,
-  id,
-  forward,
-  ...props
-}: ScrollbarProps) => {
-  // Do not remove:
-  // useMeasure ref is required for infinite scrolling to work
-  const [ref] = useMeasure({ polyfill: ResizeObserver })
-  const timerRef = useRef<NodeJS.Timeout | null>(null)
-  const elementId = useMemo(() => id || uniqueId('scrollbar-'), [id])
+export const Scrollbar = forwardRef(
+  (
+    { children, className, id, forward, isHidden, ...props }: ScrollbarProps,
+    forwardedRef: Ref<PerfectScrollbar>
+  ) => {
+    // Do not remove:
+    // useMeasure ref is required for infinite scrolling to work
+    const [ref] = useMeasure({ polyfill: ResizeObserver })
+    const timerRef = useRef<NodeJS.Timeout | null>(null)
+    const elementId = useMemo(() => id || uniqueId('scrollbar-'), [id])
 
-  useEffect(() => {
-    return () => {
+    useEffect(() => {
+      return () => {
+        if (timerRef.current !== null) {
+          clearTimeout(timerRef.current)
+        }
+      }
+    }, [])
+
+    const hideScrollbar = useCallback(() => {
+      const element = document.getElementById(elementId)
+      if (element) {
+        element.classList.remove('scrollbar--hovered-visible')
+      }
       if (timerRef.current !== null) {
         clearTimeout(timerRef.current)
       }
-    }
-  }, [])
-
-  const hideScrollbar = () => {
-    const element = document.getElementById(elementId)
-    if (element) {
-      element.classList.remove('scrollbar--hovered-visible')
-    }
-    if (timerRef.current !== null) {
-      clearTimeout(timerRef.current)
-    }
-  }
+    }, [elementId, timerRef])
 
-  const showScrollbar = () => {
-    const element = document.getElementById(elementId)
-    if (element) {
-      element.classList.add('scrollbar--hovered-visible')
-    }
-    if (timerRef.current !== null) {
-      clearTimeout(timerRef.current)
-    }
-    timerRef.current = setTimeout(() => {
+    const showScrollbar = useCallback(() => {
+      if (isHidden) return
       const element = document.getElementById(elementId)
       if (element) {
-        element.classList.remove('scrollbar--hovered-visible')
+        element.classList.add('scrollbar--hovered-visible')
       }
-    }, 1400)
-  }
+      if (timerRef.current !== null) {
+        clearTimeout(timerRef.current)
+      }
+      timerRef.current = setTimeout(() => {
+        const element = document.getElementById(elementId)
+        if (element) {
+          element.classList.remove('scrollbar--hovered-visible')
+        }
+      }, 1400)
+    }, [elementId, timerRef, isHidden])
+
+    useEffect(() => {
```

Comment:
this is the only diff in the file other than a `forwardRef` wrapper. Just this effect to call hideScrollbar on isHidden